sql/event_data_objects.cc

```diff
@@ -446,18 +465,8 @@ Event_job_data::load_from_row(THD *thd, TABLE *table)
   Event_creation_ctx::load_from_db(thd, &mem_root, dbname.str, name.str, table,
                                    &creation_ctx);
 
-  ptr= strchr(definer.str, '@');
-
-  if (! ptr)
-    ptr= definer.str;
-
-  len= ptr - definer.str;
-  definer_user.str= strmake_root(&mem_root, definer.str, len);
-  definer_user.length= len;
-  len= definer.length - len - 1;
-  /* 1:because of @ */
-  definer_host.str= strmake_root(&mem_root, ptr + 1, len);
-  definer_host.length= len;
```

Comment:
how does this change fix the issue? this seems like simple refactoring?

Reply:
Yes, it is a refactoring to eliminate some duplication. This way we eliminate future errors when we modify the logic in some part of the code but forget to update the other one.

Please, mind that we are not using `strchr` anymore but `strrcgar`

Reply:
Thanks @ldonso My eyes couldn't catch the extra "r"